merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/AutoValueMappers.java

```diff
@@ -0,0 +1,166 @@
+          autoValueMapperElement);
+    }
+
+    final var typeMirrors = new HashSet<TypeMirror>();
+    for (final var enclosedElement : autoValueMapperElement.getEnclosedElements()) {
+      if (!(enclosedElement instanceof RecordComponentElement el)) continue;
+      final var typeMirror = el.getAccessor().getReturnType();
+      typeMirrors.add(typeMirror);
+    }
+
+    return new TypeRule(
+        new TypePattern.ClassPattern(
+            ClassName.get(ValueMapper.class),
+            List.of(TypePattern.from(autoValueMapperElement.asType()))),
+        Set.of(),
+        typeMirrors
+            .stream()
+            .map(component -> (TypePattern) new TypePattern.ClassPattern(
+                ClassName.get(ValueMapper.class),
+                List.of(TypePattern.from(component).box())))
+            .toList(),
+        generatedClassName,
+        ClassName.get((TypeElement) autoValueMapperElement).canonicalName().replace(".", "_"));
+  }
+
+  static JavaFile generateAutoValueMappers(final MissionModelRecord missionModel, final Iterable<? extends Element> recordTypes) {
+    final var typeName = missionModel.getAutoValueMappersName();
+
+    final var builder =
+        TypeSpec
+            .classBuilder(typeName)
+            .addAnnotation(
+                AnnotationSpec
+                    .builder(javax.annotation.processing.Generated.class)
+                    .addMember("value", "$S", MissionModelProcessor.class.getCanonicalName())
+                    .build())
+            .addAnnotation(
+                AnnotationSpec
+                    .builder(SuppressWarnings.class)
+                    .addMember("value", "$S", "unchecked")
+                    .build())
+            .addModifiers(Modifier.PUBLIC, Modifier.FINAL);
+
+    record ComponentMapperNamePair(String componentName, String mapperName) {}
+    for (final var record : recordTypes) {
+      final var methodName = ClassName.get((TypeElement) record).canonicalName().replace(".", "_");
+      final var componentToMapperName = new ArrayList<ComponentMapperNamePair>();
+      final var necessaryMappers = new HashMap<TypeMirror, String>();
+      for (final var element : record.getEnclosedElements()) {
+        if (!(element instanceof RecordComponentElement el)) continue;
+        final var typeMirror = el.getAccessor().getReturnType();
+        final var elementName = element.toString();
+        final var valueMapperIdentifier = typeMirror
+                                              .toString()
+                                              .replace(".", "_")
+                                              .replace("<", "_")
+                                              .replace(">", "_")
+                                              .replace("[", "_")
+                                              .replace("]", "_")
+                                              .replace(",", "_")
+                                          + "ValueMapper";
```

Comment:
```suggestion
        final var valueMapperIdentifier = typeMirror
                                              .toString()
                                              .replaceAll("[.><,\[\]]", "_")
                                          + "ValueMapper";
```

Reply:
[proof](https://regex101.com/r/M0YwaZ/1)

Reply:
Thoughts on https://github.com/NASA-AMMOS/aerie/pull/850/commits/ed0ae77c4765bb8502fdc1614ee46b0b01b0bbd7 ? IntelliJ suggested that I pre-compile the regex, so I let it do that, but let me know if you think that hurts readability.

Reply:
Another thought on this - I was mildly concerned that replacing all of these with underscores _could_ result in a name collision. There isn't really any way around this in general, but I contemplated whether using different replacement characters would alleviate that risk. I think my latest stance is that the risk is low enough, and the examples contrived enough, that it's fine to replace everything with underscores